npm/private/npm_translate_lock_state.bzl

```diff
@@ -426,13 +426,26 @@ def _copy_input_file(priv, rctx, label_store, key):
         fail("key not found '{}'".format(key))
 
     if _should_update_pnpm_lock(priv):
-        # NB: rctx.read will convert binary files to text but that is acceptable for
-        # the purposes of calculating a hash of the file
-        _set_input_hash(
-            priv,
-            label_store.relative_path(key),
-            utils.hash(rctx.read(label_store.path(key))),
-        )
+        path = label_store.path(key)
+        written = False
+        if path.endswith("package.json"):
+            content = json.decode(rctx.read(path))
+            if "version" in content.keys():
+                content.pop("version")
```

Comment:
Can't we pop a lot more here? Or only read a couple key properties such as name, *dependencies? That might be a big change that should be saved for 2.0 though?

Reply:
For each thing we pop, we have to be able to reason that it shouldn't affect pnpm import. So I'd rather be conservative and think about the use cases that matter. Have you seen any other fields that are updated frequently? I'm think it's the only one on https://docs.npmjs.com/cli/v9/configuring-npm/package-json

Reply:
In my case the ones updated frequently are the package version and dependency versions that are overridden with `pnpm.overrides`. You've solved the first one, the `pnpm.overrides` one is too complicated to do here.

There are other common fields we could drop such as `scripts` it's probably not worth it since they don't change to *that* frequently.

Reply:
I think there's a good case for dropping scripts.

Reply:
Discussed over VC - the problem is that users typically also update the deps of some other package.json to point to the new version, so they're still going to have a merge conflict even with this change.